hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py

```diff
@@ -57,12 +120,39 @@ def __init__(self, key: str, value: str):
         super(self.__class__, self).__init__(key, value)
 
 
+class Action:
+    action_label: ActionLabel
+    priority: int
+    superseded_by: t.List[ActionLabel]
+
+    def __init__(self, action_label: ActionLabel, priority: int, superseded_by: t.List[ActionLabel]):
+        self.action_label = action_label
+        self.priority = priority
+        self.superseded_by = superseded_by
+
+
+class ActionRule:
+    action_label: ActionLabel
+    must_have_labels: t.List[Label]
+    must_not_have_labels: t.List[Label]
+
+    def __init__(self, action_label: ActionLabel, must_have_labels: t.List[Label], must_not_have_labels: t.List[Label]):
+        self.action_label = action_label
+        self.must_have_labels = must_have_labels
+        self.must_not_have_labels = must_not_have_labels
+
+
 def perform_action(match_message: MatchMessage, action_label: ActionLabel):
     for action, action_performer in possible_actions.items():
         if action is action_label.value:
             action_performer(match_message)
```

Comment:
Since possible_actions is a dict, you can do:
```python
possible_actions[action_label.value](match_message)
```

Looping is unnecessary and inefficient.